### PR TITLE
Network utility

### PR DESCRIPTION
**Pull request by vkedwardli**
Foundation for https://github.com/inada-s/gdxsv/issues/48, provides average ms by ping any hostname / IP address. 
```cpp
int latency = os_GetPingLatencyFromHostname("8.8.8.8"); //insert GCP IP Address here, return value's unit in ms
```

Additionally we can get the connection medium (so that we can distinguish unstable wireless players from ethernet players)
```cpp
std::string medium = std::string os_GetConnectionMedium(); //"Wired" or "Wireless"
```

Can be called directly in `gdxsv.cpp`

### CODE REVIEW

**Comment by inada-s** [comment]
There are security risks in using popen with dynamic parameter.
I think it's better to use  simply http request to mesure latency.

**Comment by vkedwardli** [comment]
Oh the `rm -rf /` attack, yes didn't think of that. Thank you for reviewing!

Changed to 
```cpp
int latency = GetLatencyFromURL("http://104.155.201.52/ping"); 
```
Placed inside `gdxsv.cpp` now (since the implementation is not os specific), feel free to delete the function, it's too simple and may not be applicable to the actual usage later.

File: core/gdxsv/gdxsv.cpp [hunk 1]
```diff
@@ -142,6 +142,7 @@ std::string Gdxsv::GeneratePlatformInfoString() {
     ss << "maxlag=" << (int) maxlag << "\n";
     ss << "patch_id=" << symbols[":patch_id"] << "\n";
     ss << "machine_id=" << os_GetMachineID() << "\n";
+    ss << "wireless=" << (os_GetConnectionMedium() == "Wireless") << "\n";
```
Comment (inada-s) [inline core/gdxsv/gdxsv.cpp]: I want it to be an integer explicitly, so please add `int` cast.

Comment (vkedwardli) [reply core/gdxsv/gdxsv.cpp]: done

**Comment by vkedwardli** [comment]
image

**Review by inada-s — APPROVED** [review]

**Pull request merged**